apps/www/registry/new-york/ui/timeline.tsx

```diff
@@ -0,0 +1,205 @@
+import React from "react"
+import { CheckIcon, CircleIcon, Cross1Icon } from "@radix-ui/react-icons"
+import { VariantProps, cva } from "class-variance-authority"
+
+import { cn } from "@/lib/utils"
+
+const timelineVariants = cva("flex flex-col items-stretch", {
+  variants: {
+    positions: {
+      left: "[&>li]:grid-cols-[0_min-content_1fr]",
+      right: "[&>li]:grid-cols-[1fr_min-content]",
+      center: "[&>li]:grid-cols-[1fr_min-content_1fr]",
+    },
+  },
+  defaultVariants: {
+    positions: "left",
+  },
+})
+
+interface TimelineProps
+  extends React.HTMLAttributes<HTMLUListElement>,
+    VariantProps<typeof timelineVariants> {}
+
+const Timeline = React.forwardRef<HTMLUListElement, TimelineProps>(
+  ({ children, className, positions, ...props }, ref) => {
+    return (
+      <ul
+        className={cn(timelineVariants({ positions }), className)}
+        ref={ref}
+        {...props}
+      >
+        {children}
+      </ul>
+    )
+  }
+)
+Timeline.displayName = "Timeline"
+
+const timelineItemVariants = cva("grid items-center gap-x-2", {
+  variants: {
+    status: {
+      done: "text-primary",
+      default: "text-muted-foreground",
+    },
+  },
+  defaultVariants: {
+    status: "default",
+  },
+})
+
+interface TimelineItemProps
+  extends React.HTMLAttributes<HTMLLIElement>,
+    VariantProps<typeof timelineItemVariants> {}
+
+const TimelineItem = React.forwardRef<HTMLLIElement, TimelineItemProps>(
+  ({ className, status, ...props }, ref) => (
+    <li
+      className={cn(timelineItemVariants({ status }), className)}
+      ref={ref}
+      {...props}
+    />
+  )
+)
+TimelineItem.displayName = "TimelineItem"
+
+const timelineDotVariants = cva(
+  "col-start-2 col-end-3 row-start-1 row-end-1 flex size-4 items-center justify-center rounded-full border border-current",
+  {
+    variants: {
+      status: {
+        default: "[&>*]:hidden",
```

Comment:
@flixlix 

First of all, thanks for preparing this component. I was using shadcn in my project and recently needed a timeline component. It was kinda urgent so until this gets merged, I wanted to give it a go by copying your component directly and everything works great!

I do have one suggestion though. I wanted to use icons from `lucide-react` package as `customIcon` and noticed it wasn't working because this line prevents all types of svgs from being displayed. Instead, I believe something like following would work better, at least it did for me.

```suggestion
        default: "[&>*:not(:nth-child(4))]:hidden",
```

Reply:
Thank you for your feedback, it is very much appreciated.

I added a variant called `custom` that shows the custom icon.

I think it makes more sense to really hide all children by default to have an "idle" status.